Interop/Runtime/src/jvm/kotlin/kotlinx/cinterop/JvmCallbacks.kt

```diff
@@ -60,9 +59,20 @@ private fun getVariableCType(type: KType): CType<*>? {
     }
 }
 
-private val structTypeCache = ConcurrentHashMap<Class<*>, CType<*>>()
+private class Caches {
+    val structTypeCache = mutableMapOf<Class<*>, CType<*>>()
+    val createdStaticFunctions = mutableMapOf<Class<*>, CPointer<CFunction<*>>>()
+}
+
+private val cachesHolder = ThreadLocal<Caches>()
+private val caches: Caches
+    get() = cachesHolder.get() ?: Caches().also { cachesHolder.set(it) }
```

Comment:
This thread local should be initialized explicitly, not on demand.
Motivation: this would help to detect bugs like accidental (and thus e.g. leaking) initialization after deinitialization.

Reply:
(same for native allocator thread local)

Reply:
This is still relevant.

Reply:
Done.